libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -893,11 +907,17 @@ ASTPointer<FunctionCall> ASTJsonImporter::createFunctionCall(Json::Value const&
 		astAssert(name.isString(), "Expected 'names' members to be strings!");
 		names.push_back(make_shared<ASTString>(name.asString()));
 	}
+
+	optional<vector<SourceLocation>> sourceLocations = createSourceLocations(_node);
+
 	return createASTNode<FunctionCall>(
 		_node,
 		convertJsonToASTNode<Expression>(member(_node, "expression")),
 		arguments,
-		names
+		names,
+		sourceLocations ?
```

Comment:
I actually did use the method with the passed size before and someone else complained about that :sweat_smile: 

About the other suggestion, it might be a bit faster, but I am not sure it would be easier to understand and maintain than a simple function returning an optional value..

Reply:
Ah, ok! 
I can agree that the optional is indeed more straightforward.